core/src/main/java/org/jruby/RubyBignum.java

```diff
@@ -236,6 +236,63 @@ public static BigInteger long2big(long arg) {
      *  ================
      */
 
+    /** rb_big_ceil
+     *
+     */
+    @Override
+    public IRubyObject ceil(ThreadContext context, IRubyObject args){
+        BigInteger ndigits = args.convertToInteger().getBigIntegerValue();
+        BigInteger self = value;
+        if (ndigits.compareTo(BigInteger.ZERO) == 1){
+            return convertToFloat();
+        } else if (ndigits.compareTo(BigInteger.ZERO) == 0){
+            return this;
+        } else {
+            BigInteger posdigits = ndigits.abs();
+            BigInteger exp = BigInteger.TEN.pow(posdigits.intValue());
+            BigInteger mod = self.mod(exp);
+            BigInteger res = self;
+            if (mod.compareTo(BigInteger.ZERO) != 0) {
+                res = self.add( exp.subtract(mod) );// self + (exp - (mod));
+            }
+            return newBignum(context.runtime, res);
+        }
+    }
+
+    /** rb_big_floor
+     *
+     */
+    @Override
+    public IRubyObject floor(ThreadContext context, IRubyObject args){
+        BigInteger ndigits = args.convertToInteger().getBigIntegerValue();
+        BigInteger self = value;
+        if (ndigits.compareTo(BigInteger.ZERO) == 1){
+            return convertToFloat();
+        } else if (ndigits.compareTo(BigInteger.ZERO) == 0){
```

Comment:
maybe `compareTo` could be performance only once and saved in a local instead of twice